### ISSUE TITLE
Soldiers not calling Taxi Service when unable to proceed on Foot

### ISSUE THREAD

**Issue by kgd192** [OpenApoc/OpenApoc#1195]
**Desktop (please complete the following information):**
 - OS:Windows 10 x64 22H2
 - OpenApoc Version 
![image](https://user-images.githubusercontent.com/15697480/233835488-c9c3a743-7ed7-4c4a-ba1c-9241200516a7.png)

**Describe the bug**
When trying to send a soldier to walk to a building, either via the alert screen or the go to building button they will not call a taxi even though the option is enabled and expected per OG behavior

**To Reproduce**
1. Load Save and fast forward and wait for Alert
2. select soldier
3. try to investigate
4. after this failed, try to send via "Go To Building Button" to the building with the alert (now centered)

**Expected behavior**
When unable to reach the target building via the peoples tube network the soldier should call a taxi and use it to reach their destination

**Savegame**
[save_agentOnFoot391.zip](https://github.com/OpenApoc/OpenApoc/files/11303486/save_agentOnFoot391.zip)

**Screenshots**
*Investigate Building*
![image](https://user-images.githubusercontent.com/15697480/233835743-8536c017-c196-4a25-a5cf-ccb76c75e0ba.png)

*go to Building manually*
![image](https://user-images.githubusercontent.com/15697480/233835771-ca9c770d-e768-4579-93f6-1492b738e856.png)

`W void __cdecl OpenApoc::AgentMission::start(class OpenApoc::GameState &,class OpenApoc::Agent &) Implement agent pathing to closest building when in the field and unable to path to building`
`W bool __cdecl OpenApoc::Agent::popFinishedMissions(class OpenApoc::GameState &) Agent Steve Brown mission "GotoBuilding BUILDING_BAKUNIN_BLOCK" finished`
`W bool __cdecl OpenApoc::Agent::popFinishedMissions(class OpenApoc::GameState &) No next agent mission, going idle`

**Comment by FilmBoy84** [comment]
@kgd192 this appears fixed by Kurtsley's PR

Can you confirm please? It seems to work with the save on my system

**Comment by kgd192** [comment]
Will recheck this today, keep you posted

**Comment by kgd192** [comment]
Yes, the being unable to proceed on foot is solved however it does not bring up the message box that all selected units have arrived at the building and whether the investigation should be started or not.

![image](https://user-images.githubusercontent.com/15697480/234349781-a14ca058-9a4b-413c-b142-22ee83e89cad.png)

**Comment by Kurtsley** [comment]
Good point will look.

**Comment by Kurtsley** [comment]
decreasePendingInvestigatorCount is not being called for agent missions like it is for vehicles. I am washing my hands of this for a while so this can be a hint for someone else to figure this out.

**Comment by kgd192** [comment]
No worries, I will have a look at it this weekend since I last modified the decreaseInvestigatorCount functionalities due to the investigator<0 error. Already have a possible solution in Mind

**Comment by kgd192** [comment]
@Kurtsley @FilmBoy84 
Just to keep you in the loop, found the root cause of the issue for not launching the investigation but just did have the time yet to think about a fix, hope to be working on it the upcoming weekend

**Comment by Artytoo** [comment]
Would a mod like Pick up Vehicle in the "8 hrs" video be a solution for the personnel stuck or crashed in city transit?

Hoverbikes could get there fast and even move them to the "final destination".

**Comment by FilmBoy84** [comment]
> Would a mod like Pick up Vehicle in the "8 hrs" video be a solution for the personnel stuck or crashed in city transit?
> 
> Hoverbikes could get there fast and even move them to the "final destination".

We actually already try to do this - IIRC, it's something like every three hours

The issue at present is that the recovery vehicles (usually construction vehicles as a type) get stuck on launch as they cannot navigate to the "stuck vehicle" so the city populates, slowly, with stuck vehicles

Now, a solution was made as a temporary workaround where vehicles stuck for around a quarter of a day would be destroyed - allowing the game to continue, but this too often fails to resolve the problem as it's always playing catch up to the number of vehicles spawning in the interim that become stuck

It would be better to resolve the largest cause of stuck vehicles - city not repairing, so tiles becoming inaccessible (See PR #1141 ) and then resolve any remnant pathfinder and destructor issues when that is finally merged (currently needs a rebase and QA check before merging)
